src/mon/MgrMonitor.cc

```diff
@@ -895,6 +900,16 @@ bool MgrMonitor::prepare_command(MonOpRequestRef op)
       r = -EINVAL;
       goto out;
     }
+    if (!pending_map.all_support_module(module)) {
+      ss << "module " << module << " does not exist";
+      r = -ENOENT;
+      goto out;
+    }
+    if (!pending_map.enabled_module(module)) {
+      ss << "module " << module << " is already disabled";
+      r = -EINVAL;
```

Comment:
This is not desired behaviour -- all Ceph CLI operations should be idempotent, so disabling a disabled module should return 0.

Reply:
fixed.